pyathena/connection.py

```diff
@@ -250,14 +254,16 @@ def __enter__(self):
     def __exit__(self, exc_type, exc_val, exc_tb):
         self.close()
 
-    def cursor(self, cursor: Optional[Type[BaseCursor]] = None, **kwargs) -> BaseCursor:
+    def cursor(self, cursor: Optional[Type[FunctionalCursor]] = None, **kwargs):
```

Comment:
```suggestion
    def cursor(self, cursor: Optional[Type[FunctionalCursor]] = None, **kwargs): -> FunctionalCursor
```
The return type is missing.

Reply:
If the `cursor` parameter is omitted, the type analyzer consistently infers `Any`.
```python
def cursor(self, cursor: Optional[Type[FunctionalCursor]] = None, **kwargs): -> FunctionalCursor:
    return cursor()

_cursor = cursor() # -> Any
```
To enhance the clarity of the return type, I recommend using a union of types
```python
def cursor(self, cursor: Optional[Type[FunctionalCursor]] = None, **kwargs): -> Union[FunctionalCursor, ConnectionCursor]:
    return cursor()

_cursor = cursor() # -> ConnectionCursor
_other_cursor = cursor(MyCursor) # -> (MyCursor | ConnectionCursor)
```
Alternatively, following your suggestion, we can ignore the generic `ConnectionCursor` type. I agree with this approach as it simplifies the code.

Reply:
I think it would be a good choice to use the Union type.

Reply:
Update Union return type 😊

Reply:
```
pyathena/connection.py:257:101: E501 Line too long (125 > 100)
```
You can format it with `make fmt`.

Reply:
The following error occurs with `make chk`.
```
$ make chk                                                                                                                                                                                                                                                                                                                                                              1043ms  Sun Jan 21 18:37:01 2024
pdm run ruff check .
pdm run ruff format --check .
42 files already formatted
pdm run mypy .
pyathena/__init__.py:60: error: Missing type parameters for generic type "Connection"  [type-arg]
pyathena/common.py:100: error: Missing type parameters for generic type "Connection"  [type-arg]
pyathena/common.py:137: error: Missing type parameters for generic type "Connection"  [type-arg]
pyathena/result_set.py:36: error: Missing type parameters for generic type "Connection"  [type-arg]
pyathena/result_set.py:43: error: Missing type parameters for generic type "Connection"  [type-arg]
pyathena/result_set.py:283: error: Missing type parameters for generic type "Connection"  [type-arg]
pyathena/result_set.py:286: error: Missing type parameters for generic type "Connection"  [type-arg]
pyathena/connection.py:72: error: Incompatible default for argument "cursor_class" (default has type "Type[Cursor]", argument has type "Type[ConnectionCursor]")  [assignment]
pyathena/connection.py:264: error: Incompatible types in assignment (expression has type "Type[ConnectionCursor]", variable has type "Type[FunctionalCursor]")  [assignment]
pyathena/pandas/util.py:138: error: Missing type parameters for generic type "Connection"  [type-arg]
pyathena/filesystem/s3.py:41: error: Missing type parameters for generic type "Connection"  [type-arg]
pyathena/arrow/result_set.py:54: error: Missing type parameters for generic type "Connection"  [type-arg]
pyathena/pandas/result_set.py:103: error: Missing type parameters for generic type "Connection"  [type-arg]
Found 13 errors in 8 files (checked 40 source files)
make: *** [chk] Error 1
```